packages/cli/src/ui/components/ModelDialog.tsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type React from 'react';
+import { useContext, useEffect, useState } from 'react';
+import { Box, Text } from 'ink';
+import {
+  DEFAULT_GEMINI_FLASH_LITE_MODEL,
+  DEFAULT_GEMINI_FLASH_MODEL,
+  DEFAULT_GEMINI_MODEL,
+  DEFAULT_GEMINI_MODEL_AUTO,
+} from '@google/gemini-cli-core';
+import { useKeypress } from '../hooks/useKeypress.js';
+import { theme } from '../semantic-colors.js';
+import { DescriptiveRadioButtonSelect } from './shared/DescriptiveRadioButtonSelect.js';
+import { ConfigContext } from '../contexts/ConfigContext.js';
+
+interface ModelDialogProps {
+  onClose: () => void;
+  onSelect: (model: string) => void;
+}
+
+const MODEL_OPTIONS = [
+  {
+    value: DEFAULT_GEMINI_MODEL_AUTO,
+    title: 'Auto (recommended)',
+    description: 'Let the system choose the best model for your task',
+  },
+  {
+    value: DEFAULT_GEMINI_MODEL,
+    title: 'Pro',
+    description: 'For complex tasks that require deep reasoning and creativity',
+  },
+  {
+    value: DEFAULT_GEMINI_FLASH_MODEL,
+    title: 'Flash',
+    description: 'For tasks that need a balance of speed and reasoning',
+  },
+  {
+    value: DEFAULT_GEMINI_FLASH_LITE_MODEL,
+    title: 'Flash-Lite',
+    description: 'For simple tasks that need to be done quickly',
+  },
+];
+
+export function ModelDialog({
+  onClose,
+  onSelect,
+}: ModelDialogProps): React.JSX.Element {
+  const config = useContext(ConfigContext);
+  const [selectedModel, setSelectedModel] = useState(
+    config?.getModel() || DEFAULT_GEMINI_MODEL_AUTO,
+  );
+
+  useEffect(() => {
```

Comment:
I think this useEffect is a no-op. you are already passing in 
config?.getModel in the useState. 
Are you concerned about the case of the config changing while the dialog is open? Config should be fixed today so no need to check for that... probably have bigger problems if the config changes while the model dialog is open.